### ISSUE TITLE
Pylint runs unexpectedly pass if `confidence=` in pylintrc

### ISSUE THREAD

**Issue by jacobtylerwalls** [PyCQA/pylint#6512]
### Bug description

Runs unexpectedly pass in 2.14 if a pylintrc file has `confidence=`.

(Default pylintrc files have `confidence=`. `pylint`'s own config was fixed in #6140 to comment it out, but this might bite existing projects.)

```python
import time
```

### Configuration

```ini
[MESSAGES CONTROL]
confidence=
```

### Command used

```shell
python3 -m pylint a.py --enable=all
```

### Pylint output

```shell
--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)
```

### Expected behavior
```
************* Module a

a.py:2:0: C0305: Trailing newlines (trailing-newlines)
a.py:1:0: C0114: Missing module docstring (missing-module-docstring)
a.py:1:0: W0611: Unused import time (unused-import)

--------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 10.00/10, -10.00)
```
### Pylint version

```shell
pylint 2.14.0-dev0
astroid 2.12.0-dev0
Python 3.10.2 (v3.10.2:a58ebcc701, Jan 13 2022, 14:50:16) [Clang 13.0.0 (clang-1300.0.29.30)]
```

**Comment by jacobtylerwalls** [comment]
The documentation of the option says "Leave empty to show all."

**Comment by DanielNoord** [comment]
```diff
diff --git a/pylint/config/argument.py b/pylint/config/argument.py
index 8eb6417dc..bbaa7d0d8 100644
--- a/pylint/config/argument.py
+++ b/pylint/config/argument.py
@@ -44,6 +44,8 @@ _ArgumentTypes = Union[
 
 def _confidence_transformer(value: str) -> Sequence[str]:
     """Transforms a comma separated string of confidence values."""
+    if not value:
+        return interfaces.CONFIDENCE_LEVEL_NAMES
     values = pylint_utils._check_csv(value)
     for confidence in values:
         if confidence not in interfaces.CONFIDENCE_LEVEL_NAMES:
```

This will fix it.

I do wonder though: is this worth breaking? Probably not, but it is counter-intuitive and against all other config options behaviour to let an empty option mean something different. `confidence` should contain all confidence levels you want to show, so if it is empty you want none. Seems like a bad design choice when we added this, perhaps we can still fix it?...

**Comment by jacobtylerwalls** [comment]
Thanks for the speedy reply! I don't think we should bother to change it. It's clearly documented, so that's something!

**Comment by DanielNoord** [comment]
Hm okay. Would you be willing to prepare a PR with the patch? I had intended not to spend too much time on `pylint` this evening 😄
